lib/utils.js

```diff
@@ -115,7 +116,41 @@ function getManifest(options) {
 exports.getManifest = getManifest;
 
 function getXPIManifest(xpiPath) {
+  return inspectXPI(xpiPath)
+    .then(function(tmpXPI) {
+      var manifest = {};
+      try {
+        manifest = require(path.join(tmpXPI.path, "package.json"));
+      } catch (e) {}
+
+      tmpXPI.remove();
+
+      return manifest;
+    });
+}
+
+/**
+ * Takes an `xpiPath` (e.g. filename.xpi), extracts it, and resolves
+ * the returned promise with an object to let you work with the
+ * temporary directory.
+ *
+ * The resolution object has the following attributes:
+ * - `path`: temporary directory path containing the XPI contents.
+ * - `remove()`: method that removes the temporary directory. You must
+ *   call this explicitly when you are finished.
+ *
+ * @param {Object} tmpXPI
+ * @return {Promise}
+ */
+function inspectXPI(xpiPath) {
```

Comment:
heh, I had it named that originally. Names are hard. My thinking was that since it's temporary you do extract it but the main purpose is for inspecting. I can change it back though, doesn't really make sense now that I try to explain it.